origin-growth/src/apollo/schema.js

```diff
@@ -0,0 +1,96 @@
+const { gql } = require('apollo-server-express')
+
+const typeDefs = gql`
+  scalar JSON
+
+  ######################
+  #
+  # Query output schema.
+  #
+  ######################
+
+  enum CampaignStatus {
+    pending                   #not yet started
+    active
+    capReached
+    completed
+  }
+
+  enum ActionStatus {
+    inactive
+    active
+    exhausted
+    completed
+  }
+
+  enum ActionType {
+    basicProfile
+    attestations
+    referral
+  }
+
+  enum InviteStatus {
+    pending
+    successful
+  }
+
+  type Price {
+    currency: String!
+    amount: String!
+  }
+
+  type Invite {
+    status: InviteStatus!
+    walletAddress: ID!
+    contactName: String
+    reward: Price
+  }
+
+  interface BaseAction {
+    type: ActionType!
+    status: ActionStatus!
+    rewardEarned: Price
+    rewardPending: Price      # applicable for referral action
+    rewardInfo: Price!        # information about reward
+  }
+
+  type Action implements BaseAction {
+    type: ActionType!
+    status: ActionStatus!
+    rewardEarned: Price
+    rewardPending: Price      # applicable for referral action
+    rewardInfo: Price!        # information about reward
+  }
+
+  type ReferralAction implements BaseAction {
+    type: ActionType!
+    status: ActionStatus!
+    rewardEarned: Price
+    rewardPending: Price      # applicable for referral action
+    rewardInfo: Price!        # information about reward
+    invites: [Invite]         # TODO: might we need pagination for invites?
```

Comment:
Yes, might want to add pagination based on this model used by DApp2: https://github.com/OriginProtocol/origin/blob/master/experimental/origin-graphql/src/typeDefs/Marketplace.js#L122

Reply:
Agreed, also proposed by the [graphQL pagination tutorial](https://graphql.org/learn/pagination/)